### DISCUSSION TITLE
Linearlite - live queries might not reflect update on drag and drop???

### DISCUSSION THREAD

**Discussion by morgs32** [electric-sql/electric#1400] [General]
I see this [comment here](https://github.com/electric-sql/electric/blob/6002f07ff8d940c4176c0edab09667e005300cd7/examples/linearlite/src/pages/Board/IssueBoard.tsx#L23):

```
  // Issues are coming from a live query, this may not have updated before we rerender
  // after a drag and drop. So we keep track of moved issues and use that to override
  // the status of the issue when sorting the issues into columns.
```

Why would this be? Is it in the edge case where something else causes this page to rerender?? Like props? 
Or is it because the 16ms that it should take to update is not fast enough??

**Comment by samwillis** [comment]
Hey @morgs32 

This is due to the react drag and drop lib being fully synchronous, whereas the Electric API is async. When you drop the tile it synchronously updates state in the d&d lib which causes an immediate re-render, but the call to persist the change to the database is async. So even though the update is under the 16ms for a frame it's pushed to the next as it's scheduled in a micro task. It's a side effect of mixing async and sync state.

There are a few ways to mitigate this, one we use here were track the moves and invalidate a dirty state, or to use a fully synchronous build of SQLite in the main thread.

**Reply by morgs32** [reply]
That makes sense thanks.

**Comment by KyleAMathews** [comment]
👋 we've been working the last month on a rebuild of the Electric server over at a temporary repo https://github.com/electric-sql/electric-next/

You can read more about why we made the decision at https://next.electric-sql.com/about

We're really excited about all the new possibilities the new server brings and we hope you'll check it out soon and give us your feedback.

We're now moving the temporary repo back here. As part of that migration we're closing all the old issues, PRs, and discussions. We really appreciate you taking the time to start this discussion! If it's still relevant in the new system, please reopen.
